### PR TITLE
CB-13362. Enable pre-flight network checks event history notification…

### PR DESCRIPTION
**Pull request by oleewere**
…s (+ remove entitlement)

See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java [hunk 1]
```diff
@@ -69,16 +77,16 @@
...
             if (rpcResponse != null) {
                 LOGGER.debug("Diagnostics network report response: {}", rpcResponse.getFirstTextMessage());
-                if (rpcResponse.getResult() != null && useNotifications) {
-                    List<NodeStatusProto.NetworkDetails> networkNodes = rpcResponse.getResult().getNodesList().stream()
+                NodeStatusProto.NodeStatusReport nodeStatusReport = rpcResponse.getResult();
+                if (nodeStatusReport != null) {
+                    String cdpTelemetryVersion = nodeStatusReport.getCdpTelemetryVersion();
+                    boolean stableNetworkCheckSupported = isVersionGreaterOrEqual(cdpTelemetryVersion, STABLE_NETWORK_CHECK_VERSION);
+                    boolean awsMetadataServerV2Supported = isVersionGreaterOrEqual(cdpTelemetryVersion, AWS_METADATA_SERVER_V2_SUPPORT_VERSION);
+                    List<NodeStatusProto.NetworkDetails> networkNodes = nodeStatusReport.getNodesList().stream()
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java:89]: Nit: potential NPE if `getNodesList() == null`.

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java [hunk 2]
```diff
@@ -297,4 +328,10 @@ private boolean allNetworkNodesInUnknownStatus(List<NodeStatusProto.NetworkDetai
                 .collect(Collectors.toList());
     }
 
+    public boolean isVersionGreaterOrEqual(String actualVersion, String versionToCompare) {
+        ModuleDescriptor.Version actVersion = ModuleDescriptor.Version.parse(actualVersion);
+        ModuleDescriptor.Version versionToCmp = ModuleDescriptor.Version.parse(versionToCompare);
+        return actVersion.compareTo(versionToCmp) >= 0;
+    }
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java:335]: This certainly works, just wondering if using the CB `VersionComparator` would be better for consistency.

**Review by lajosrodek — APPROVED** [review]
LGTM, only two minor comments.

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java (hunk 2 above)
Comment (oleewere) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java:335]: i would rather use this, as it works for different version formats and the calculation won't change if the CB versioning changes for some reason

**Pull request merged**